gateway/ln-gateway/src/lightning/lnd.rs

```diff
@@ -629,6 +633,132 @@ impl ILnRpcClient for GatewayLndClient {
             failure_reason: "Gatewayd has not started to route HTLCs".to_string(),
         })
     }
+
+    async fn connect_to_peer(
+        &self,
+        pubkey: secp256k1::PublicKey,
+        host: String,
+    ) -> Result<EmptyResponse, LightningRpcError> {
+        let mut client = Self::connect(
+            self.address.clone(),
+            self.tls_cert.clone(),
+            self.macaroon.clone(),
+        )
+        .await?;
```

Comment:
This is repeated a lot, maybe worth to create a helper fn?

Reply:
Done. Created #4924 and rebased this PR on top of it.